### PR TITLE
DC-4783: imi endpoint to send and view emailQueue

### PR DESCRIPTION
**Pull request by muralishub**

### CODE REVIEW

File: app/uk/gov/hmrc/digitalcontactstub/repositories/EmailQueueRepository.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    ) {
+  def save(emailContent: EmailContent): Future[Boolean] =
+    collection.insertOne(emailContent).toFuture().map(_.wasAcknowledged())
+
+  def cleanUp: Future[Boolean] =
+    collection.deleteMany(Filters.empty).toFuture().map(_.wasAcknowledged())
+
+  def findAll = collection.find(Filters.empty).toFuture()
+  def findItem(id: String) =
+    collection.find(Filters.eq("to.correlationId", id)).toFuture()
```
Comment (ardwalker) [inline app/uk/gov/hmrc/digitalcontactstub/repositories/EmailQueueRepository.scala:43]: You might want to think about indexing on this

File: app/uk/gov/hmrc/digitalcontactstub/service/EmailQueueService.scala [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+
+  def addToQueue(emailContent: EmailContent)(
+      implicit ec: ExecutionContext): Future[EmailQueued] = {
+    for {
+      _ <- emailQueueRepository.save(emailContent)
+      queued = EmailQueued(timeStamp,
+                           UUID.randomUUID().toString,
+                           emailContent.to.map(_.correlationId).head,
+                           "queued")
```
Comment (ardwalker) [inline app/uk/gov/hmrc/digitalcontactstub/service/EmailQueueService.scala:42]: Might consider making this a type

File: project/CodeCoverageSettings.scala [hunk 3]
```diff
@@ -16,7 +16,7 @@ object CodeCoverageSettings {
 
   val settings: Seq[Setting[_]] = Seq(
     ScoverageKeys.coverageExcludedPackages := excludedPackages.mkString(";"),
-    ScoverageKeys.coverageMinimumStmtTotal := 100,
+    ScoverageKeys.coverageMinimumStmtTotal := 50,
```
Comment (ardwalker) [inline project/CodeCoverageSettings.scala:19]: Coverage is a bit on the low side

File: scalastyle-config.xml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ <name>Scalastyle standard configuration</name>
+ <check level="warning" class="org.scalastyle.file.FileTabChecker" enabled="true"></check>
+ <check level="warning" class="org.scalastyle.file.FileLengthChecker" enabled="true">
+  <parameters>
+   <parameter name="maxFileLength"><![CDATA[800]]></parameter>
+  </parameters>
+ </check>
+ <check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="true">
+  <parameters>
+   <parameter name="header"><![CDATA[/*
```
Comment (ardwalker) [inline scalastyle-config.xml:11]: Do we need to specify a file header here, doesn't the `sbt-auto-build` plugin produce one for us?

File: project/CodeCoverageSettings.scala (hunk 3 above)
Comment (muralishub) [reply project/CodeCoverageSettings.scala:19]: this is a stub , do you think its better to keep 80 or 90% like we do for other services

File: scalastyle-config.xml (hunk 4 above)
Comment (muralishub) [reply scalastyle-config.xml:11]: this is the same we do for the other services

File: project/CodeCoverageSettings.scala (hunk 3 above)
Comment (ardwalker) [reply project/CodeCoverageSettings.scala:19]: Yes, if its just a stub, I would not worry about it.

File: scalastyle-config.xml (hunk 4 above)
Comment (ardwalker) [reply scalastyle-config.xml:11]: Don’t  worry for now, but I had a conversation with plat-ops the other day and it sounds like that's what happens. We should discuss as a team and perhaps clean up some of these configuration files if necessary.

**Review by ardwalker — APPROVED** [review]

File: app/uk/gov/hmrc/digitalcontactstub/repositories/EmailQueueRepository.scala (hunk 1 above)
Comment (muralishub) [reply app/uk/gov/hmrc/digitalcontactstub/repositories/EmailQueueRepository.scala:43]: we deal with less data for testing, we can add in future if needed

**Pull request merged**